### PR TITLE
add UI for proposal review workflow

### PR DESCRIPTION
**Pull request by jashan-lco**
https://github.com/observatorycontrolsystem/observation-portal/pull/299

### CODE REVIEW

**Comment by jashan-lco** [comment]
@jnation3406 Since you reviewed the API changes, could I also get your eyes on this? It's deployed to https://obs-portal-review-process.preview.lco.earth/proposal-reviews with a clone DB

**Comment by nvolgenau** [comment]
The link to the PDF of a proposal yields a message that begins "This XML file does not appear to have any style information associated with it. The document tree is shown below." An example error message that follows:

<Error>
<Code>NoSuchKey</Code>
<Message>The specified key does not exist.</Message>
<Key>new_media/sciapps/2024B/20331/SJ_LCO24B__Watanabe.pdf</Key>
<RequestId>GT95JFYQW4WFA312</RequestId>
<HostId>CKsSeYzjz3E1j7cTCzsnQ38EBnJR1uA/nGJz/XNEnmhywfU6mRHB61YypjjC9K9ass8AfBdJX2w=</HostId>
</Error>

This error occurs both from the "Proposal Reviews" table and from the individual "my_review" pages.

**Comment by nvolgenau** [comment]
Question. On the "change science application review" admin page (e.g. https://obs-portal-review-process.preview.lco.earth/admin/sciapplications/scienceapplicationreview/35/change/ ), is the "Notify submitter additional message" field for the body of Lisa's email to the PI? Will Lisa have to cut-and-paste the same text into this field for each proposal? Can this field include a table displaying hours allocated?

Question. When is the email sent to the PI? (When the "Notify sumbitter" box is checked, and the page is Saved?) BTW, "sumbitter" -> "submitter".

**Comment by nvolgenau** [comment]
"Exoplanets" should be added to the list of Science categories.

File: src/views/ProposalReviewUserDetail.vue [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        processData: "false",
+        contentType: "application/json",
+        cache: false
+      }).done(function(response) {
+        that.apiValidationErrors = {};
+
+        that.data.my_review = response;
+        that.$store.commit("addMessage", { text: "Saved", variant: "success" });
+
+        that.dataNotFound = false;
```
Comment (jnation3406) [inline src/views/ProposalReviewUserDetail.vue:144]: This is up for debate but I kind of think after saving it should route you back to the proposal reviews table page... At least when you save after selecting finished maybe?

**Review by jnation3406 — APPROVED** [review]
Looks good!

**Comment by jashan-lco** [comment]
> The link to the PDF of a proposal yields a message that begins "This XML file does not appear to have any style information associated with it. The document tree is shown below." An example error message that follows:
> `NoSuchKey` The specified key does not exist. new_media/sciapps/2024B/20331/SJ_LCO24B__Watanabe.pdf GT95JFYQW4WFA312 CKsSeYzjz3E1j7cTCzsnQ38EBnJR1uA/nGJz/XNEnmhywfU6mRHB61YypjjC9K9ass8AfBdJX2w=
> 
> This error occurs both from the "Proposal Reviews" table and from the individual "my_review" pages.

This is expected because it's running in the dev environment without access to production keys to access PDF files which are stored in another service (S3).

**Comment by jashan-lco** [comment]
> Question. On the "change science application review" admin page (e.g. https://obs-portal-review-process.preview.lco.earth/admin/sciapplications/scienceapplicationreview/35/change/ ), is the "Notify submitter additional message" field for the body of Lisa's email to the PI? Will Lisa have to cut-and-paste the same text into this field for each proposal? Can this field include a table displaying hours allocated?
> 

Yes, this text will be embedded here, if not empty: https://github.com/observatorycontrolsystem/observation-portal/blob/b2fa7da11771c26f6e23254139161ff18b923dc2/observation_portal/sciapplications/templates/sciapplications/review_accepted_or_rejected.txt#L7

We discussed generating this message automatically based on other information available, but this seems like a good first step. If you can get me some samples of what you'd like, I can try to implement that.

> Question. When is the email sent to the PI? (When the "Notify sumbitter" box is checked, and the page is Saved?) BTW, "sumbitter" -> "submitter".

Email is sent anytime the status is changed to 'Accepted' or 'Rejected'. This implies you can send the email multiple times if you change the status back and forth (by accident or on purpose). But the "Notify submitter" must be checked before the status change (or at the same time). I could change this so that it's checked by default, if that's a more common scenario.

File: src/views/ProposalReviewUserDetail.vue (hunk 1 above)
Comment (jashan-lco) [reply src/views/ProposalReviewUserDetail.vue:144]: Going to keep it simple for now. If people find it cumbersome to navigate back and forth, we can add it then.

**Pull request merged**